### ISSUE TITLE
Curious about your GPU specification

### ISSUE THREAD

**Issue by thynl** [jwkirchenbauer/lm-watermarking#10]
Hello, thanks for your great work!
While reading the paper, I became curious about your experimental setting.
I wanted to know the GPU type that you used for the experiment, but couldn't find it in the paper...
May I ask you the type of GPU that you used, and how long did it take for text completion generation (500 samples?) in your environment?
Thanks!!

**Comment by jwkirchenbauer** [comment]
Thanks for the interest!
For all experiments (to parallelize the hparam sweep) we used a single RTXA4/5000 or RTXA6000. For 8 way beam search, the larger card was required as the search occurs in the batch dimension (see the huggingface implementation of beamsearch for details). The generation times were around a few hours, but I don't have exact times easily accessible.

---
Extra notes:

Primary reasons for this runtime was that we run the 1.3B and 2.7B models in full precision and oversample - i.e. generate between 1000-2000 samples, and then subselect only the 500 that have sufficient length (~200) to compute statistics. 

The other reason was that we ran them unbatched - one generation per model call (except beam search which implicitly creates a batch). However, the new processor logic should support batching just fine, so we expect that the generations can be sampled faster. That said, the watermarking logic itself isn't naturally parallelizable because we use the standard pytorch random number generator as our seeding function, and this isn't a vectorized operation. Additionally, the tensor permutation to select the greenlist is also not vectorized - `torch.randperm` can't permute a batch of tensors for you, so we have to select greenlists one at a time in the batch dimension.
